### PR TITLE
Partner platform/component pull request

### PR DESCRIPTION
**Pull request by betzw**
Prepare handover to @bcostm

Substitutes #17 and adjusts request from @0xc0170 regarding version number handling etc.

Group of related PRs:
- https://github.com/ARMmbed/mbed-hal-st-stm32f4/pull/25
- https://github.com/ARMmbed/cmsis-core-stm32f4/pull/18
- https://github.com/ARMmbed/mbed-hal-st-stm32f401re/pull/4
- https://github.com/ARMmbed/cmsis-core-nucleo-f401re/pull/4
- https://github.com/ARMmbed/target-st-nucleo-f401re-gcc/pull/4

### CODE REVIEW

File: module.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   },
   "targetDependencies": {
     "stm32f401xe": {
-      "cmsis-core-stm32f401xe": "~0.2.0"
+      "cmsis-core-stm32f401xe": "~0.0.1"
```
Comment (0xc0170) [inline module.json]: Current version is v0.1.0, I am not able to publish. @bcostm is, can he publish it, we update this one?

Comment (betzw) [inline module.json]: What should the version number be in order to allow you to merge it into master?

**Comment by 0xc0170** [comment]
I tested all outstanding PR for 401 nucleo locally, there are just those ver errors I reported, it compiles and runs the blinky, we will have to run tests in the main repositories

File: module.json (hunk 1 above)
Comment (0xc0170) [inline module.json]: v0.1.0, but should be published, to make CI green

**Comment by betzw** [comment]
Ok @0xc0170, all version errors should be fixed!

**Comment by 0xc0170** [comment]
Merged manually

**Pull request closed without merging**